### PR TITLE
Add support for starting with devices

### PR DESCRIPTION
**Pull request by bigfish24**
This is my first attempt to add support for starting a VM with physical devices.

    1. I added a new key in the yaml: `start-devices` which accepts an array of device names/IDs
    2. I added a check that if `start-devices is present to ensure like the modify VM commands to stop/suspend the VM
    3. I added a step after the VM is stopped to start it with the `-d <NAME> -d <NAME..` option(s)
    4. I added a test trying to verify this.

However, my test is failing and I am really not sure why. I want to share for feedback on how to fix.

### CODE REVIEW

**Comment by bigfish24** [comment]
@NorseGaud could you take a look at this?

File: tests/command.bats [hunk 1]
```diff
@@ -3,8 +3,8 @@
 load '/usr/local/lib/bats/load.bash'
 
 # Uncomment to enable stub debug output:
-# export ANKA_STUB_DEBUG=/dev/tty
-# export BUILDKITE_PLUGIN_ANKA_DEBUG=true
+export ANKA_STUB_DEBUG=/dev/tty
```
Comment (NorseGaud) [inline tests/command.bats]: I think these need to be commented out -- chef guys can confirm

Comment (bigfish24) [reply tests/command.bats]: I will but my test is failing and I can’t figure out why, any ideas?

Comment (NorseGaud) [reply tests/command.bats]: BATS testing is really a pain. When you enable /dev/tty, you need to stub each and every anka command that shows. Try adding this:

```
  stub anka \
    "list ${BUILDKITE_PLUGIN_ANKA_VM_NAME} : exit 0" \
    "clone ${BUILDKITE_PLUGIN_ANKA_VM_NAME} ${BUILDKITE_PLUGIN_ANKA_VM_NAME}-${BUILDKITE_JOB_ID} : echo cloned vm in anka" \
    "list ${BUILDKITE_PLUGIN_ANKA_VM_NAME}-${BUILDKITE_JOB_ID} : echo suspended grep" \
    "stop --force ${BUILDKITE_PLUGIN_ANKA_VM_NAME}-${BUILDKITE_JOB_ID} : echo stopped anka vm" \
    "start -d iphone1 -d iphone2 ${BUILDKITE_PLUGIN_ANKA_VM_NAME}-${BUILDKITE_JOB_ID} : echo started with devices" \
    "run ${BUILDKITE_PLUGIN_ANKA_VM_NAME}-${BUILDKITE_JOB_ID} bash -c \"ls -alht\" : echo ran ls command in anka" \

  run $PWD/hooks/command

  assert_output --partial "cloned vm in anka"
  assert_output --partial "stopped anka vm"
  assert_output --partial "started with devices"
  assert_output --partial "ran ls command in anka"

  assert_success
```

Also, do you mind updating the docker-compose for me to:
```
version: '3.4'
services:
  tests:
    image: buildkite/plugin-tester
    volumes:
      - ".:/plugin"
```
I've been meaning to get to it.

**Comment by tduffield** [comment]
@bigfish24 Thanks for the contribution. @NorseGaud is right about the bats and the README modifications. We'll also need you to DCO sign off your commit(s) when things are sorted out.

File: tests/command.bats (hunk 1 above)
Comment (bigfish24) [reply tests/command.bats]: thanks @NorseGaud, I got it working from this!

**Comment by bigfish24** [comment]
Tested this with our infrastructure and it is working properly!

**Comment by NorseGaud** [comment]
@bigfish24 thanks for doing this!

**Pull request merged**